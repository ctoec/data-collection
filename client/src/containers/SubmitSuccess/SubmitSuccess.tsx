import React from 'react';
import { useParams } from 'react-router-dom';
import moment from 'moment';
import { MailToLink } from '../../components/MailToLink';
import {
  Button,
  TextWithIcon,
  ArrowRight as Arrow,
} from '@ctoec/component-library';
import { CSVDownloadLink } from '../../components/CSVDownloadLink';
import LogoWithCheckSrc from '@ctoec/component-library/dist/assets/images/logoWithCheck.svg';

import { Link } from 'react-router-dom';

/**
 * TODO: Right now, the CheckData page just redirects straight
 * here when you click the Send to OEC button. Once we have the
 * infrastructure for what it means to actually send to OEC,
 * we can alter this, but for now, we need a way to actually
 * make this page aware of what data was submitted.
 */
const SubmitSuccess: React.FC = () => {
  // Any cast when extracting param suppresses type error
  // Fine because we know what the IDs are
  const { reportId } = useParams();
  // const idString: string = rawIds['idArray'];

  return (
    <div className="grid-container margin-top-4">
      <div className="grid-row text-bold">
        <Link to="/check-data">
          <TextWithIcon
            text="Back"
            Icon={Arrow}
            direction="left"
            iconSide="left"
          />
        </Link>
      </div>
      <div className="tablet:grid-col-auto margin-top-2 margin-bottom-2">
        <img src={LogoWithCheckSrc} alt="" />
      </div>
      <div>
        <h1>You've submitted your data to OEC!</h1>
        <p className="margin-top-1 text-bold">
          Date submitted: {moment.utc().format('MM/DD/YYYY')}
        </p>
      </div>
      <div className="grid-row">
        <p>
          We'll let you know if any edits are required.
          <br />
          If you have any questions, reach out to
          <br /> <MailToLink />
        </p>
      </div>
      <div className="grid-row margin-top-2">
        <strong>Want a copy of your enrollment data?</strong>
      </div>
      <div className="grid-row margin-top-1 margin-bottom-2">
        <CSVDownloadLink reportId={reportId} />
      </div>
      <div className="grid-row marging-top-6">
        <strong>Need to make changes?</strong>
      </div>
      <div className="grid-row margin-top-1">
        <Button text="Upload another file" href="/upload" />
      </div>
    </div>
  );
};

export default SubmitSuccess;
