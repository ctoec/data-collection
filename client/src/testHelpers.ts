import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';

export const accessibilityTestHelper = (
  component: React.ReactElement,
  before?: (_: RenderResult) => Promise<void>
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = render(component);
      if (before) {
        await before(renderResult);
      }
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });
    cleanup();
  });
};

export const snapshotTestHelper = (component: React.ReactElement) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const { asFragment } = render(component);
      expect(asFragment()).toMatchSnapshot();
    });
  });
};
