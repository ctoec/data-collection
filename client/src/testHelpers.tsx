import React from 'react';
import { RenderResult, render, cleanup, act } from '@testing-library/react';
import { axe } from 'jest-axe';
import { BrowserRouter } from 'react-router-dom';

export const renderHelper = async (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  const renderResult = render(
    opts.wrapInRouter ? <BrowserRouter>{component}</BrowserRouter> : component
  );

  if (opts.before) {
    await opts.before(renderResult);
  }
  return renderResult;
};

export const accessibilityTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('passes AXE accessibility checks', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      const results = await axe(renderResult.container);
      expect(results).toHaveNoViolations();
    });

    cleanup();
  });
};

export const snapshotTestHelper = (
  component: React.ReactElement,
  opts: {
    before?: (_: RenderResult) => Promise<void>;
    wrapInRouter?: boolean;
  } = {}
) => {
  it('matches snapshot', async () => {
    await act(async () => {
      const renderResult = await renderHelper(component, opts);
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    cleanup();
  });
};
