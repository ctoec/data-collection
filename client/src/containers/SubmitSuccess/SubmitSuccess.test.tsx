import React from 'react';
import { snapshotTestHelper, accessibilityTestHelper } from '../../testHelpers';
import SubmitSuccess from './SubmitSuccess';

describe('SubmitSuccess', () => {
  snapshotTestHelper(<SubmitSuccess />, { wrapInRouter: true });
  accessibilityTestHelper(<SubmitSuccess />, { wrapInRouter: true });
});
