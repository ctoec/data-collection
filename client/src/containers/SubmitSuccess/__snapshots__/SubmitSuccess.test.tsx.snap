// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitSuccess matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container margin-top-4"
  >
    <div
      class="grid-row text-bold"
    >
      <a
        class="usa-button usa-button--unstyled"
        href="/roster"
      >
        <span
          class="oec-text-with-icon"
        >
          Back to Roster
        </span>
      </a>
    </div>
    <div
      class="tablet:grid-col-auto margin-top-2 margin-bottom-2"
    >
      <img
        alt=""
        src="logoWithCheck.svg"
      />
    </div>
    <div>
      <h1>
        You've submitted your data to OEC!
      </h1>
      <p
        class="margin-top-1 text-bold"
      >
        Date submitted: 06/22/2008
      </p>
    </div>
    <div
      class="grid-row"
    >
      <p>
        We'll let you know if any edits are required.
        <br />
        If you have any questions, reach out to
        <br />
         
        <a
          class="usa-button usa-button--unstyled"
          href="mailto:oec-data-pilot@skylight.digital?subject=Data%20collection%20bug"
          rel="noopener noreferrer"
          target="_blank"
        >
          oec-data-pilot@skylight.digital
        </a>
      </p>
    </div>
    <div
      class="grid-row margin-top-2"
    >
      <strong>
        Want a copy of your enrollment data?
      </strong>
    </div>
    <div
      class="grid-row margin-top-1 margin-bottom-2"
    >
      <button
        class="usa-button usa-button--unstyled margin-bottom-3 display-block"
        type="button"
      >
        <span
          class="oec-text-with-icon text-underline"
        >
          Export roster
        </span>
      </button>
    </div>
    <div
      class="grid-row marging-top-6"
    >
      <strong>
        Need to make changes?
      </strong>
    </div>
    <div
      class="grid-row margin-top-1"
    >
      <a
        class="usa-button"
        href="/upload"
      >
        Upload another file
      </a>
    </div>
  </div>
</DocumentFragment>
`;
